src/cargo/ops/cargo_new.rs

```diff
@@ -925,3 +941,79 @@ fn update_manifest_with_inherited_workspace_package_keys(
         try_remove_and_inherit_package_key(key, manifest);
     }
 }
+
+/// Adds the new package member to the [workspace.members] array.
+/// - It first checks if the name matches any element in [workspace.exclude],
+///  and it ignores the name if there is a match.
+/// - Then it check if the name matches any element already in [workspace.members],
+/// and it ignores the name if there is a match.
+/// - If [workspace.members] doesn't exist in the manifest, the new package
+/// is not added to the members array. This prevents manifest files from breaking
+/// if they're only used for package templating.
+fn update_manifest_with_new_member(
+    manifest_path: &Path,
+    workspace_document: &mut toml_edit::Document,
+    relative_path: &str,
+) -> CargoResult<()> {
+    // Don't add the new package to the workspace's members
+    // if there is an exclusion match for it.
+    if let Some(members) = workspace_document
+        .get("workspace")
+        .and_then(|workspace| workspace.get("exclude"))
+        .and_then(|members| members.as_array())
+    {
+        let members = members.iter().map(|s| s.as_str().unwrap());
+        if matches_in_package_list(members, relative_path)? {
+            return Ok(());
+        }
+    }
+
+    if let Some(members) = workspace_document
+        .get("workspace")
+        .and_then(|workspace| workspace.get("members"))
+        .and_then(|members| members.as_array())
+    {
+        let members = members.iter().map(|s| s.as_str().unwrap());
+        if !matches_in_package_list(members, relative_path)? {
+            let Some(toml_members) = workspace_document["workspace"]["members"].as_array_mut()
+            else {
+                anyhow::bail!("[workspace.members] is not an array");
+            };
+            toml_members.push(relative_path);
```

Comment:
What formatting situations should we detect and preserve?

Reply:
I can't think of any examples but to dig into `toml_edit`s API.

There are likely to main formats to work with
```toml
members = ["one", "two"]
members = [
    "one",   # amount of indentation may vary
    "two",  # use of trailing comma may vary
]
```

With `cargo add`, the only format preserving we do is to keep the list of dependencies sort if it isn't.  We are prescriptive on how to format dependencies rather than making it configurable.

Here, we might have some wiggle room if the people working on `Cargo.toml` style guide (@joshtriplett ?) want to be prescriptive on indentation or trailing commas, because we can just detect specific cases and preserve those and let the user deal with it otherwise.

A third route is a "we don't care" and leave it to a future version of `cargo fmt` that updates `Cargo.toml`.  This is basically the approach that `cargo fix` takes.

Reply:
... that is strange.  I'm the maintainer of `toml_edit` and I'm not sure why that worked.

Reply:
With the idea that we will have `Cargo.toml` auto-fix in the near future, I think it'd be worthwhile doing what auto-fix does for Rust: punt to `cargo fmt`.  Going to check with others to see if we're ok moving forward with that.

Reply:
See https://rust-lang.zulipchat.com/#narrow/stream/246057-t-cargo/topic/Automatic.20edits.20to.20.60Cargo.2Ctoml.60